### ISSUE TITLE
val loss is 0

### ISSUE THREAD

**Issue by 1326320543** [WongKinYiu/yolov9#132]
I am training YOLOV9 with box loss, cls loss, and dfl loss values of 0 for the three val, and it is not clear what is causing the results to be 0

**Comment by WongKinYiu** [comment]
Currently calculation of val loss is closed when use `train_dual.py`.
https://github.com/WongKinYiu/yolov9/blob/main/val_dual.py#L194

**Comment by roman-forever** [comment]
hello ，when I open the calculation of val loss in train_dual.py, but an error was reported because the image size did not match. So what should I do?

**Comment by shahriarahmadf** [comment]
> Currently calculation of val loss is closed when use `train_dual.py`. https://github.com/WongKinYiu/yolov9/blob/main/val_dual.py#L194

Can you tell how to open that calculation of val_loss during train_dual.py?
Results graph having the val_loss, and other plots not becoming zero is better to evaluate overfitting/underfitting. Please respond.

**Comment by hu694** [comment]
so can I uncomment code line 195 in val_dual?

**Comment by orel26** [comment]
has anyone tried to uncomment code line 194 in val_dual? does it work now?

**Comment by PFangwang** [comment]
> has anyone tried to uncomment code line 194 in val_dual? does it work now?

i tried but failed,it does't work,i don't know why...Should I open the comments on these two lines
        # Loss
        if compute_loss:
            preds = preds[1]
            #train_out = train_out[1]
            #loss += compute_loss(train_out, targets)[1]  # box, obj, cls

**Comment by prammmoe** [comment]
have a same issue here, try to uncomment those lines but still doesn't work XD

**Comment by babakbch** [comment]
Same problem here, I uncommented those lines and still get 0, is this resolved? am I missing something?
Any help is appreciated

**Comment by loe2022** [comment]
> > has anyone tried to uncomment code line 194 in val_dual? does it work now?
> 
> i tried but failed,it does't work,i don't know why...Should I open the comments on these two lines # Loss if compute_loss: preds = preds[1] #train_out = train_out[1] #loss += compute_loss(train_out, targets)[1] # box, obj, cls

do you slove this now?can this work?I meet the same problem but I can't slove it.

**Comment by loe2022** [comment]
has anyone slove this?Any help is appreciated

**Comment by naruto123sdo** [comment]
> has anyone slove this?Any help is appreciated

@loe2022 You can simply uncomment the lines computing validation loss.

But what's the motivation disabling this in train_dual.py? Is it inaccurate? @WongKinYiu

**Comment by pranavbudhwant** [comment]
> has anyone slove this?Any help is appreciated

```
if compute_loss:
    preds = preds[1]
    #train_out = train_out[1]
    loss += compute_loss(train_out, targets)[1]  # box, obj, cls
```

This works @loe2022 (keep [L196]( https://github.com/WongKinYiu/yolov9/blob/5b1ea9a8b3f0ffe4fe0e203ec6232d788bb3fcff/val_dual.py#L196) commented)
